pecan/compat/__init__.py

```diff
@@ -18,3 +18,38 @@
 
 def is_bound_method(ob):
     return inspect.ismethod(ob) and six.get_method_self(ob) is not None
+
+
+def __get_argspec__():
+    from collections import namedtuple
+    return namedtuple('ArgSpec', 'args varargs keywords defaults')
+
+ArgSpec = __get_argspec__()
+def getargspec(func):
+    if six.PY2:
+        return inspect.getargspec(func)
+
+    sig = inspect._signature_from_callable(func, follow_wrapper_chains=False,
```

Comment:
I wish I could claim credit for this code, but it was mostly lifted from a commit in Django:

https://github.com/timgraham/django/commit/3872a33132a4bb6aa22b237927597bbfdf6f21d7